app/src/main/java/cz/martykan/forecastie/weatherapi/WeatherStorage.java

```diff
@@ -0,0 +1,146 @@
+package cz.martykan.forecastie.weatherapi;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.util.Log;
+
+import androidx.annotation.Nullable;
+
+import org.json.JSONException;
+
+import java.util.List;
+
+import cz.martykan.forecastie.Constants;
+import cz.martykan.forecastie.models.Weather;
+import cz.martykan.forecastie.weatherapi.owm.OpenWeatherMapJsonParser;
+
+public class WeatherStorage {
+    protected SharedPreferences sharedPreferences;
+
+    public WeatherStorage(Context context) {
+        this.sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+    }
+
+    public int getCityId()
+    {
+        String cityIdString = sharedPreferences.getString("cityId", Constants.DEFAULT_CITY_ID);
+        return Integer.parseInt(cityIdString);
+    }
+
+    public void setCityId(int cityId)
+    {
+        // TODO: Add migration
+        String cityIdString = Integer.toString(cityId);
+        sharedPreferences.edit().putString("cityId", cityIdString).apply();
```

Comment:
I believe the to-do can be removed, as no migration is needed if we keep `"cityId"` as a string.

Reply:
yep, go for it! i think it should be a string

Reply:
Fixed in 3629f50fbee9acd5ff180c741355bb80471d1f6b